proposal-EDGI_ML_project.md

```diff
@@ -0,0 +1,156 @@
+
+Location: Nagpur, Maharashtra, India
+
+Indian Standard Time(IST), GMT +05:30
+
+GitHub handle: janakrajchadha
+
+Slack handle: the\_automator
+
+Email: janakrajchadha@gmail.com
+
+Other Contact: Linkedin - [https://goo.gl/ePFmSs](https://goo.gl/ePFmSs)
+
+Twitter - [@the\_automator\_](https://twitter.com/the_automator_)
+
+### **Project Details**
+
+**Project Title** : Identifying and prioritizing important website changes using learning algorithms.
+
+**Project Abstract/Summary** (&lt;20 words): The project goal is to utilise machine learning algorithms to identify changes on government agency websites which are worth reviewing.
+
+**Describe the need your project fills** :
+
+The ignorant and disparaging stand of the new U.S administration on environmental issues like global warming, air pollution etc. has led to panic as federal agencies gradually make the ocean of scientific data on government servers inaccessible to the public. The project will attempt to tackle the aforementioned issue and fill the following needs:
+
+1. Filter out insignificant changes (which may be a major share of all changes) to webpages, and thus, eliminate the need to spend time reviewing useless information.
+2. Direct analysts to important changes on active websites which will help them to efficiently document and analyse relevant information.
+3. Identify missing data and information files and attempt to find archived versions on other websites.
+
+**Describe how your project meets this need** :
+
+The changes worth reviewing will tend to have some characteristic features which can be classified with the help of various classification algorithms. A considerable amount of time will be devoted to the preparation of the training dataset which is an essential task while working on Machine Learning problems.
+
+A variety of Machine Learning models will be trained on the thoroughly labeled dataset. They will also be tested to evaluate the performance by using some part of the data for validation.
+
+The project will follow the following order of tasks to fill the needs:
+
+1. Step zero will be converting data from different sources (PageFreezer, Versionista) into a common form and storing them in one place.
+2. The first step will be improving the method of computing differences. This will consist of two sub-tasks:
+  1. The changes can be categorized based on the type of object that has changed (text, image, hyperlinks etc.). The first sub-task will be to separate the changes into categories.
+  2. The second sub-task will be to extract meaningful features from the data of each type. This will include extracting information like type and length of the changed matter in a given text. If required, complex techniques can be used to extract features (NLP, Image Processing).
+3. The second task will be writing scripts for different Machine Learning Classification Algorithms which will learn to prioritize the changes. Each script will consist of four basic parts:
```

Comment:
Similarly this task seems like a big (and important) one! 
I'm wondering if @mhucka and @danielballan have thoughts around how you could work on these in an iterative way and check in with analysts throughout these steps?